src/rgw/rgw_op.cc

```diff
@@ -5867,14 +5867,15 @@ void RGWCopyObj::execute(optional_yield y)
     if (op_ret < 0) {
       return;
     }
+    obj_size = s->src_object->get_size();
 
     /* Check if the src object is cloud-tiered */
     bufferlist bl;
     if (s->src_object->get_attr(RGW_ATTR_MANIFEST, bl)) {
       RGWObjManifest m;
       try{
         decode(m, bl);
-        if (m.is_tier_type_s3()) {
+        if (m.is_tier_type_s3() && (obj_size == 0)) {
```

Comment:
nit: please add a comment that non-zero size represents restored copy

Reply:
sure